lib/view_model/active_record/controller.rb

```diff
@@ -62,12 +62,64 @@ def destroy(serialize_context: new_serialize_context, deserialize_context: new_d
   end
 
   included do
-    etag { self.viewmodel_class.deep_schema_version }
+    etag { migrated_deep_schema_version }
+  end
+
+  def parse_viewmodel_updates
+    super.tap do |update_hash, refs|
+      if migration_versions.present?
+        migrator = ViewModel::UpMigrator.new(migration_versions)
+        migrator.migrate!([update_hash, refs], references: refs)
+      end
+    end
+  end
+
+  def prerender_viewmodel(*)
+    super do |jbuilder|
+      yield(jbuilder) if block_given?
```

Comment:
Consider `defined?(yield)` here for performance, as this is likely a pretty frequented path.

https://medium.com/rubycademy/the-defined-keyword-in-ruby-b7a5a5a48e1e

Reply:
1: Wow, that's amazing, Ruby never fails to surprise with new lows in its implementation. That `defined?`, a method-like thing that interrogates the current binding is different from `block_given?`, a method-like thing that interrogates the current binding is quite a 🦜 .

2: It's not a particularly hot path, since it's only called once per controller method; I think I'd rather go with the more idiomatic version to be less surprising.

Reply:
Agreed that it isn't needed, but I think the difference is that `block_given?` is a method, and `defined?` is a keyword.